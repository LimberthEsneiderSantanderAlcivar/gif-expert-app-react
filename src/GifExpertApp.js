import { useState } from "react"
import AddCategory from "./components/AddCategory"
import GifGrid from "./components/GifGrid"

const GifExpertApp = () => {
    
    const [categories, setcategories] = useState(['Shingeki No Kyojin'])
   
    return (
        <>
            <h2>GifExpertApp</h2>  
            <AddCategory setcategories={setcategories}/>
            <hr/>


            <ol>   
                {
                    categories.map((cat) => 
                        <GifGrid 
                            key={cat}
                            category={cat}
                        />
                    )
                }
            </ol> 
        </>
    )
}
export default GifExpertApp;
