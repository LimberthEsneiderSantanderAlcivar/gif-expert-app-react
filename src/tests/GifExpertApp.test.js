import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('preubas del componetne GifExpertApp', () => {
    test('debe mostrar el compoennte corectamente', () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe mostrar una liusta de categorias', () => {
        const categories = ['one punch', 'dragon ball'];

        const wrapper = shallow (<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
    
})
