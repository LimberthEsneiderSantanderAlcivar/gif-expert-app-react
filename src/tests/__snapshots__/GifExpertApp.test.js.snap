// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preubas del componetne GifExpertApp debe mostrar el compoennte corectamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setcategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`preubas del componetne GifExpertApp debe mostrar una liusta de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setcategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="one punch"
      key="one punch"
    />
    <GifGrid
      category="dragon ball"
      key="dragon ball"
    />
  </ol>
</Fragment>
`;
